File: smart_farm/src/api/operationLog.ts
import { get } from '@/utils/request'
import type { PageInfo } from '@/api/user'

export interface OperationLog {
  id: number
  operatorId?: number
  operatorName?: string
  operation?: string
  ip?: string
  operateTime?: string
}

export interface OperationLogListParams {
  operatorName?: string
  operation?: string
  startTime?: string
  endTime?: string
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export const listOperationLogs = (params: OperationLogListParams) =>
  get<ApiResult<PageInfo<OperationLog>>>('/operation-log/list', params as unknown as Record<string, unknown>)

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/OperationLog.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OperationLog {

    private Long id;

    private Long operatorId;

    private String operatorName;

    private String operation;

    private String ip;

    private LocalDateTime operateTime;
}
